coap-core/src/main/java/com/mbed/coap/transport/udp/DatagramSocketTransport.java

```diff
@@ -72,30 +76,32 @@ public DatagramSocketTransport(int localPort) {
     }
 
     public void setSocketBufferSize(int socketBufferSize) {
-        if (socket != null) {
+        if (socketCreated()) {
             throw new IllegalStateException();
         }
         this.socketBufferSize = socketBufferSize;
     }
 
     public void setReuseAddress(boolean reuseAddress) {
-        if (socket != null) {
+        if (socketCreated()) {
             throw new IllegalStateException();
         }
         this.reuseAddress = reuseAddress;
     }
 
+
     @Override
     public void start(CoapReceiver coapReceiver) throws IOException {
-        if (socket == null) {
-            socket = createSocket();
+        if (!socketCreated()) {
+            createSocket();
         }
 
         if (socketBufferSize > 0) {
-            socket.setReceiveBufferSize(socketBufferSize);
-            socket.setSendBufferSize(socketBufferSize);
+            updateReceiveBufferSize();
+            updateSendBufferSize();
```

Comment:
Lets remove that from here. Setting buffer sizes could be done in actual implementation of datgramsocket.

Reply:
fixed.